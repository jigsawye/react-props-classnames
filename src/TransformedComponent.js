import PropTypes from 'prop-types';

import PropsTransform from './PropsTransform';

const TransformedComponent = ({
  options,
  classNames,
  children,
  ...otherOrops
}) => {
  const propsTransform = new PropsTransform(options);
  const propsClassNames = propsTransform.getClassNamesFromProps(otherOrops);
  const mergedClassNames = `${classNames} ${propsClassNames}`.trim();

  return children({ classNames: mergedClassNames, ...otherOrops });
};

TransformedComponent.displayName = 'PropsTransform';

TransformedComponent.propTypes = {
  classNames: PropTypes.string,
  options: PropTypes.shape({
    string: PropTypes.string,
    bool: PropTypes.bool,
    prefix: PropTypes.string,
  }),
};

TransformedComponent.defaultProps = {
  classNames: '',
  options: {},
};

export default TransformedComponent;
