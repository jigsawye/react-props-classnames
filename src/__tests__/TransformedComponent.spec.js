import React from 'react';
import { shallow } from 'enzyme';

import TransformedComponent from '../TransformedComponent';

describe('TransformedComponent', () => {
  it('should export TransformedComponent', () => {
    expect(TransformedComponent).toBeDefined();
  });

  it('should render TransformedComponent', () => {
    const component = shallow(
      <TransformedComponent disabled size="lg">
        {props => <div {...props}>Test</div>}
      </TransformedComponent>
    );

    expect(component).toMatchSnapshot();

    expect(component.prop('className')).toEqual(
      'default-prefix-disabled default-prefix-size-lg'
    );
  });
});
