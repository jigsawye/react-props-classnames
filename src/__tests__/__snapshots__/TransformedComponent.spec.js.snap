// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformedComponent should render TransformedComponent 1`] = `
<div
  classNames="default-prefix-disabled default-prefix-size-lg"
  disabled={true}
  size="lg"
>
  Test
</div>
`;
